## Proses Pemberian Judul Take Home Test Project

Prediksi Potensi Pembelian Pada Pelanggan E-Commerce Berdasarkan Aktivitas Pengguna.

## Proses Business Problem
1. Suatu perusahaan e-commerce memiliki banyak data pengguna yang melakukan berbagai aktivitas seperti pencarian produk, melihat produk, membaca ulasan produk, dan menambahkan produk ke keranjang. Tetapi dari semua aktivitas tersebut, tidak selalu berakhir menjadi sebuah transaksi pembelian.

2. Perusahaan membutuhkan sistem yang mampu mengidentifikasi potensi pembelian sebuah produk berdasarkan pola aktivitas pengguna yang dilakukan selama menggunakan aplikasi e-commerce.

## Proses Problem Statement
1. Bagaimana membangun model machine learning dan deep learning yang mampu memprediksi kemungkinan pelanggan melakukan pembelian berdasarkan aktivitas pengguna yang dilakukan di aplikasi e-commerce?

## Proses Business Objective

1. Mengidentifikasi faktor yang mempengaruhi pembelian.

2. Membangun model prediksi pembelian.

3. Membantu tim marketing melakukan strategi promosi yang lebih tepat sasaran.

## Proses Import Library



In [34]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

pd.set_option("display.max_columns", None)

## Proses Load Dataset

In [35]:
data_pelanggan = pd.read_excel(
    r"D:\Dokumen Bootcamp Dibimbing 2025\[WAJIB] Final Take Home Test\Take Home Test_Raihan Azhar Rafi\Data\final_customer_data.xlsx"
)

data_pelanggan.head(10)

,User_id,Session_id,DateTime,Category,SubCategory,Action,Quantity,Rate,Total Price
0,52243841613,d76fde-8bb3-4e00-8c23,2019-01-10 10:20:00,Electronic Appliances,Speakers,first_app_open,NaN,NaN,NaN
1,52243841613,33dfbd-b87a-4708-9857,2019-01-10 10:22:00,Electronic Appliances,Speakers,search,NaN,NaN,NaN
2,57314161118,6511c2-e2e3-422b-b695,2019-01-10 14:00:00,Men's Fashion,Jeans,search,NaN,NaN,NaN
3,57314161118,90fc70-0e80-4590-96f3,2019-01-10 14:07:00,Men's Fashion,Jeans,product_view,NaN,NaN,NaN
4,57314161118,bd7419-2748-4c56-95b4,2019-01-10 14:12:00,Men's Fashion,Jeans,read_reviews,NaN,NaN,NaN
5,59221776934,0d91c2-c9c2-4e81-90a5,2019-08-10 17:00:00,Mobile & Accessories,Mobile,add_review,NaN,NaN,NaN
6,51629142904,e811e9-91de-46da-90c3,2019-09-03 12:09:00,Cleaning supplies,Cleaning sprays,search,NaN,NaN,NaN
7,51629142904,srtdf7-25c8-4147-901d,2019-09-03 12:12:00,Cleaning supplies,Cleaning sprays,product_view,NaN,NaN,NaN
8,51629142904,vfgscx-91de-4da-90c4,2019-09-03 12:22:00,Cleaning supplies,Cleaning sprays,search,NaN,NaN,NaN
9,51629142904,hgfhgf7-25c8-417-901d,2019-09-03 14:29:00,Cleaning supplies,Cleaning sprays,product_view,NaN,NaN,NaN


## Insight Proses Load Dataset

1. Dataset berisi aktivitas pelanggan pada platform e-commerce.
2. Data ini akan digunakan untuk memahami perilaku pelanggan dan membangun model prediksi pembelian.

## Proses Melihat Dimensi Dataset

In [36]:
data_pelanggan.shape

(2090, 9)

## Insight Proses Melihat Dimensi Dataset

1. Dataset memiliki 2090 baris data.
2. Dataset memiliki 9 kolom.
3. Dari data baris dan kolom tersebut, sudah cukup untuk melakukan eksplorasi data dan membangun model machine learning.

## Proses Melihat Informasi Dataset

In [37]:
data_pelanggan.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2090 entries, 0 to 2089
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   User_id      2090 non-null   int64  
 1   Session_id   2090 non-null   object 
 2   DateTime     2090 non-null   object 
 3   Category     2090 non-null   object 
 4   SubCategory  2090 non-null   object 
 5   Action       2090 non-null   object 
 6   Quantity     201 non-null    float64
 7   Rate         201 non-null    float64
 8   Total Price  201 non-null    float64
dtypes: float64(3), int64(1), object(5)
memory usage: 147.1+ KB


## Insight Proses Melihat Informasi Dataset

1. Terdapat beberapa tipe data yaitu int64, float64, dan object.
2. Kolom Quantity, Rate, dan Total Price memiliki jumlah data yang lebih sedikit dibanding kolom lainnya. Hal tersebut menunjukkan bahwa terdapat indikasi missing value yang harus ditangani pada tahap preprocessing.

## Proses Ringkasan Dataset

In [38]:
ringkasan_dataset = pd.DataFrame({
    "Informasi": [
        "Jumlah Pelanggan",
        "Jumlah Fitur",
        "Target"
    ],
    "Nilai": [
        data_pelanggan.shape[0],
        data_pelanggan.shape[1]-1,
        "target_pembelian"
    ]
})

ringkasan_dataset

,Informasi,Nilai
0,Jumlah Pelanggan,2090
1,Jumlah Fitur,8
2,Target,target_pembelian


## Proses Pemeriksaan Missing Value

In [39]:
data_pelanggan.isnull().sum()

User_id           0
Session_id        0
DateTime          0
Category          0
SubCategory       0
Action            0
Quantity       1889
Rate           1889
Total Price    1889
dtype: int64

## Insight Proses Pemeriksaan Missing Value

1. Missing value ditemukan pada kolom Quantity, Rate, dan Total Price sebanyak 1889 data.
2. Selanjutnya dilakukan investigasi lebih lanjut untuk mengetahui penyebab missing value tersebut.

## Proses Analisis Missing Value pada Kolom Quantity

In [40]:
data_pelanggan[
    data_pelanggan["Quantity"].isnull()
]["Action"].value_counts()

Action
read_reviews           342
search                 320
product_view           292
add_to_cart            280
checkout               202
click_wishlist_page    202
first_app_open         121
add_review              69
add_to_wishlist         33
product_review          22
add_to_wishist           5
read_review              1
Name: count, dtype: int64

## Insight Proses Analisis Missing Value pada Kolom Quantity

1. Missing value paling banyak ditemukan pada aktivitas read_reviews(342 data), search(320 data), dan product_view(292 data).
2. Tidak ditemukan aktivitas purchase pada daftar missing value. Hal tersebut menunjukkan bahwa data transaksi selalu tersedia ketika pembelian terjadi.
3. Berarti, missing value kemungkinan muncul disaat terdapat sebuah aktivitas yang belum melakukan proses transaksi.


## Proses Analisis Missing Value pada Kolom Rate

In [41]:
data_pelanggan[
    data_pelanggan["Rate"].isnull()
]["Action"].value_counts()

Action
read_reviews           342
search                 320
product_view           292
add_to_cart            280
checkout               202
click_wishlist_page    202
first_app_open         121
add_review              69
add_to_wishlist         33
product_review          22
add_to_wishist           5
read_review              1
Name: count, dtype: int64

## Insight Proses Analisis Missing Value pada Kolom Rate

1. Dapat dilihat bahwa pola missing value pada kolom Rate sama seperti kolom Quantity yang sudah di analisis sebelumnya.
2. Pada aktivitas non-transaksi tidak memiliki informasi harga produk. Hal tersebut menunjukkan bahwa case dari missing value ini berasal dari proses bisnis yang normal.

## Proses Analisis Missing Value pada Kolom Total Price

In [42]:
data_pelanggan[
    data_pelanggan["Total Price"].isnull()
]["Action"].value_counts()

Action
read_reviews           342
search                 320
product_view           292
add_to_cart            280
checkout               202
click_wishlist_page    202
first_app_open         121
add_review              69
add_to_wishlist         33
product_review          22
add_to_wishist           5
read_review              1
Name: count, dtype: int64

## Insight Proses Analisis Missing Value pada Kolom Total Price

1. Kurang lebih hasil analisis nya sama dengan yang terjadi di kolom Quantity dan Rate, yaitu missing value hanya muncul pada aktivitas pelanggan yang belum menghasilkan transaksi.
2. Sehingga bisa dilanjutkan ke tahap berikutnya yaitu proses menangani missing value.

## Proses Menangani Missing Value

In [43]:
data_pelanggan["Quantity"] = (
    data_pelanggan["Quantity"]
    .fillna(0)
)

data_pelanggan["Rate"] = (
    data_pelanggan["Rate"]
    .fillna(0)
)

data_pelanggan["Total Price"] = (
    data_pelanggan["Total Price"]
    .fillna(0)
)

data_pelanggan.isnull().sum()

User_id        0
Session_id     0
DateTime       0
Category       0
SubCategory    0
Action         0
Quantity       0
Rate           0
Total Price    0
dtype: int64

## Insight Proses Menangani Missing Value

1. Missing value pada Quantity, Rate, dan Total Price diubah menjadi 0.
2. Nilai 0 sengaja dipilih karena aktivitas pada 3 kolom tersebut belum menghasilkan transaksi.

## Proses Pemeriksaan Data Duplikat

In [44]:
data_pelanggan.duplicated().sum()

0

## Proses Statistik Deskriptif

In [45]:
data_pelanggan.describe()

,User_id,Quantity,Rate,Total Price
count,2.090000e+03,2090.000000,2090.000000,2090.000000
mean,5.383871e+10,0.285167,686.023923,909.344498
std,2.781474e+09,2.382323,5033.290983,6287.728594
min,5.144382e+10,0.000000,0.000000,0.000000
25%,5.176871e+10,0.000000,0.000000,0.000000
50%,5.220572e+10,0.000000,0.000000,0.000000
75%,5.566042e+10,0.000000,0.000000,0.000000
max,6.030264e+10,100.000000,90000.000000,160000.000000


## Insight Proses Statistik Deskriptif

1. Informasi ini berguna untuk memahami karakteristik data transaksi pelanggan yang berisi ringkasan data numerik berupa (nilai minimum, maksimum, rata-rata, dan standar deviasi).
2. Hasil statistik tersebut akan digunakan sebagai dasar analisis pada tahap EDA.